packages/reputation-miner/ReputationMiner.js

```diff
@@ -520,6 +520,60 @@ class ReputationMiner {
     return [retBranchMask, siblings];
   }
 
+  /**
+   * Get a Merkle proof and value for `key` in a past reputation state with root hash `rootHash`
+   * @param  {[type]}  rootHash A previous root hash of a reputation state
+   * @param  {[type]}  key      A key in that root hash we wish to know the value and proof for
+   * @return {Promise}          A promise that resolves to [branchmask, siblings, value] for the supplied key in the supplied root hash
+   */
+  async getHistoricalProofAndValue(rootHash, key) {
+    const tree = new patriciaJs.PatriciaTree();
+    // Load all reputations from that state.
+
+    const db = await sqlite.open(this.dbPath, { Promise });
+
+    let res = await db.all(
+      `SELECT reputations.skill_id, reputations.value, reputation_states.root_hash, colonies.address as colony_address, users.address as user_address
+       FROM reputations
+       INNER JOIN colonies ON colonies.rowid=reputations.colony_rowid
+       INNER JOIN users ON users.rowid=reputations.user_rowid
+       INNER JOIN reputation_states ON reputation_states.rowid=reputations.reputation_rowid
+       WHERE reputation_states.root_hash="${rootHash}"`
+    );
+    if (res.length === 0) {
+      return new Error("No such reputation state");
+    }
+    for (let i = 0; i < res.length; i += 1) {
+      const row = res[i];
+      const rowKey = await ReputationMiner.getKey(row.colony_address, row.skill_id, row.user_address); // eslint-disable-line no-await-in-loop
+      await tree.insert(rowKey, row.value); // eslint-disable-line no-await-in-loop
+    }
+
+    const keyElements = ReputationMiner.breakKeyInToElements(key);
+    const [colonyAddress, , userAddress] = keyElements;
+    const skillId = new BN(keyElements[1]).toString();
+    res = await db.all(
+      `SELECT reputations.value
+      FROM reputations
+      INNER JOIN colonies ON colonies.rowid=reputations.colony_rowid
+      INNER JOIN users ON users.rowid=reputations.user_rowid
+      INNER JOIN reputation_states ON reputation_states.rowid=reputations.reputation_rowid
+      WHERE reputation_states.root_hash="${rootHash}"
+      AND users.address="0x${userAddress}"
+      AND reputations.skill_id="${skillId}"
+      AND colonies.address="0x${colonyAddress}"`
+    );
+
+    if (res.length === 0) {
+      return new Error("No such reputation");
+    }
```

Comment:
I guess we could. It should never be more than one though, because of the check done [when saving the state](https://github.com/JoinColony/colonyNetwork/blob/develop/packages/reputation-miner/ReputationMiner.js#L804).